packages/app-mobile/services/e2ee/NativeEncryption.react-native.ts

```diff
@@ -0,0 +1,51 @@
+import { NativeEncryptionInterface } from '@joplin/lib/services/e2ee/types';
+import crypto from 'react-native-quick-crypto';
+import { HashAlgorithm } from 'react-native-quick-crypto/lib/typescript/keys';
+
+class NativeEncryption implements NativeEncryptionInterface {
+	public getCiphers(): string[] {
+		return crypto.getCiphers();
+	}
+
+	public getHashes(): string[] {
+		return crypto.getHashes();
+	}
+
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any -- TODO: remove the type "any" here
+	public async randomBytes(size: number): Promise<Buffer | any> {
```

Comment:
Here are a few possible solutions:
1. Make `NativeEncryptionInterface.randomBytes` return a `Promise<BufferLike>`, where `BufferLike` is defined to be a subset of `node:Buffer`'s methods. For example,
   ```ts
   interface BufferLike {
       arrayBuffer(): ArrayBuffer;
       slice(a: number, b: number): BufferLike;
       toString(encoding: string): string;
       // ... other properties/methods that are used by the encryption code ...
   }
   ````
2. Return an `ArrayBuffer`, `Uint8Array`, or similar.

Reply:
I will try them in the next few days. Thanks!

Reply:
I think it's fine to use the `any` now and maybe we can change it to a better type definition when the native encryption part is finished.

Reply:
I've created type `CryptoBuffer` for the incompatible variables. It extends `Uint8Array` with an extra method `toString()`. I will add other methods if necessary.